src/app/app.component.ts

```diff
@@ -31,49 +32,41 @@ export class AppComponent implements OnInit, OnDestroy {
   }
 
   changeCharactersInput(element: any): void {
-    // 1.1. Add functionality to changeCharactersInput method. Changes searchTermByCharacters Subject value on input change.
     const inputValue: string = element.target.value;
-    // YOUR CODE STARTS HERE
-
-    // YOUR CODE ENDS HERE
+    this.searchTermByCharacters.next(inputValue);
   }
 
   initCharacterEvents(): void {
-    // 1.2. Add API call on each user input. Use mockDataService.getCharacters - to make get request.
-
-    // 2. Since we don't want to spam our service add filter by input value and do not call API until a user enters at least 3 chars.
-
-    // 3. Add debounce to prevent API calls until user stop typing.
-
     this.charactersResults$ = this.searchTermByCharacters
-        .pipe
-        // YOUR CODE STARTS HERE
-
-        // YOUR CODE ENDS HERE
-        ();
+        .pipe(
+          map((input: string) => (input ? input.trim() : "")),
+          filter(input => input.length >= 3),
+          debounceTime(500),
+          switchMap((input) => this.mockDataService.getCharacters(input))
+        );
   }
 
   loadCharactersAndPlanet(): void {
-    // 4. On clicking the button 'Load Characters And Planets', it is necessary to process two requests and combine the results of both requests into one result array. As a result, a list with the names of the characters and the names of the planets is displayed on the screen.
-    // Your code should looks like this: this.planetAndCharactersResults$ = /* Your code */
-    // YOUR CODE STARTS HERE
-    // YOUR CODE ENDS HERE
+    this.planetAndCharactersResults$ = forkJoin([
```

Comment:
Hi Rositsa! I was assigned to check your home task. I think this method missing some parts, can you check it again?

Reply:
I missed some parts trying to understand this new conception.. :D I will try to fixed it :)

Reply:
The forkJoin will return un array with the combine results from the 2 APIs, I am missing this part. I should subscribe it to load that data?

Reply:
I looked at your salution just to try to understand. I hope you do not mind. In your case you are using it as [forkJoin] with a dictionary of observable inputs and you map it so you can receive an array. In my case I am using [forkJoin] with an array of observable inputs, so as result I get an array with the data of those. So my question is should I map it?